src/quicktype-core/language/CPlusPlus.ts

```diff
@@ -940,9 +1048,111 @@ export class CPlusPlusRenderer extends ConvenienceRenderer {
     }
 
     protected emitHelperFunctions(): void {
+
+        if (this._options.codeFormat) {
+            this.emitBlock(["class ClassMemberConstraint"], true, () => {
```

Comment:
Actually, these class and member names have to be `Name`s, conforming to the naming convention selected in the options.  Goes for `minValue`, etc, too.

Reply:
Hmm help me with this. What should I call to generate the correct names?

Reply:
Actually, for the members of this class you don't have use a `Name`, since there can't be any conflict with other names, but you will have to use the name styling function selected by the current options.

Actually, come to think of it, you have two options for `ClassMemberConstraint`, and any other top-level name you introduce (the problem here, apart from name style, is that they can conflict with generated types):

1. Make a `Name` for them.  We've never needed this yet, so there's no "nice" way to do it.  You'll have to overload `setUpNaming()`, call it on `super`, and then add your own name.  Make sure you return the value that the call to `super` returned.

2. Easier: You can add the properly styled named dynamically to the forbidden names.  You just need to add it to the array returned by `forbiddenNamesForGlobalNamespace()`.  I think this is what you should do.